titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/topology/DefaultTopologyFactory.java

```diff
@@ -146,15 +149,20 @@ private static int getJobMaxSkew(Job<?> job) {
         }
 
         DisruptionBudgetPolicy policy = job.getJobDescriptor().getDisruptionBudget().getDisruptionBudgetPolicy();
+        // Job spreading is only relevant for jobs that care about the availability.
         if (!(policy instanceof AvailabilityPercentageLimitDisruptionBudgetPolicy)) {
```

Comment:
minor, can we make this part of the `DisruptionBudgetPolicy` contract? It will make this behavior more robust against new disruption policies being added in the future (e.g. I will soon add another `GangMigration` policy and will most definitely not remember to come check this logic):

```diff
diff --git a/titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/DisruptionBudgetPolicy.java b/titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/DisruptionBudgetPolicy.java
index d7acaa646..fb5513bc9 100644
--- a/titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/DisruptionBudgetPolicy.java
+++ b/titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/DisruptionBudgetPolicy.java
@@ -17,4 +17,5 @@
 package com.netflix.titus.api.jobmanager.model.job.disruptionbudget;

 public abstract class DisruptionBudgetPolicy {
+    public abstract boolean isJobSpreadingRequired();
 }
```

Reply:
We could, but we agreed that the spreading is a concept that has nothing to do with the disruption budget.
I added this check here now, but it could change to something else in the future.
I am not totally opposed, but let's think a bit more about that.